Expensify/App
# Order by most recent along with report icons

~HOLD for Web-E [PR](https://github.com/Expensify/Web-Expensify/pull/29599) to go live.~
~HOLD for Web-E [PR](https://github.com/Expensify/Web-Expensify/pull/29707) to go live~
HOLD for Web-E [PR](https://github.com/Expensify/Web-Expensify/pull/29768) to go live.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/145238

### Tests
1. Pull in this web-E [PR](https://github.com/Expensify/Web-Expensify/pull/29599).
1. Open expensify chat. Click through a few of your reports. Remember the order of reports you clicked through if possible.
![image](https://user-images.githubusercontent.com/14356145/98854433-686cc400-240f-11eb-99ae-7aa9428a22e0.png)
3. Click on the search bar. Confirm you see your reports sorted in the same order.
![image](https://user-images.githubusercontent.com/14356145/98854578-981bcc00-240f-11eb-80a2-661cdf3389f2.png)
4. Confirm 1:1 DMs show up with Add button and group DMs don't have add button.
1. Click on a name and confirm that report opens up.
1. Click search again and type out a name. This is not sorted by last visited but we confirm that nothing is broken here.
1. Select a name or add to group chat and confirm nothing is broken.
1. Confirm in the LHN, recently visited chats or while chat searching that any user that has an icon shows their icons.
![image](https://user-images.githubusercontent.com/14356145/98856843-0615c280-2413-11eb-8095-54dfabd826c8.png)

### Screenshots
Pinned chats (the only difference is that icons now show up here as well)
![image](https://user-images.githubusercontent.com/14356145/98854433-686cc400-240f-11eb-99ae-7aa9428a22e0.png)

Pinned chats mobile
![image](https://user-images.githubusercontent.com/14356145/98854461-702c6880-240f-11eb-9cf2-b8be4f66ce85.png)

Empty chat search showing recently visited chats - web
![image](https://user-images.githubusercontent.com/14356145/98854578-981bcc00-240f-11eb-80a2-661cdf3389f2.png)

Empty chat search showing recently visited chats - mobile
![image](https://user-images.githubusercontent.com/14356145/98854495-791d3a00-240f-11eb-98cf-91bf30ec7e3e.png)

Searching for a person - web
![image](https://user-images.githubusercontent.com/14356145/98854620-a4078e00-240f-11eb-921e-c4ce01b4f4bf.png)

Searching for a person - mobile
![image](https://user-images.githubusercontent.com/14356145/98854646-acf85f80-240f-11eb-9168-c89ef97befc5.png)
